packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -22,6 +22,15 @@ function throwIfAborted (signal) {
   if (signal?.aborted) { throw createAbortError('The operation was aborted', { cause: signal.reason }) }
 }
 
+function abortable (promise, signal) {
+  const abortPromise = () => promise.abort(signal.reason)
+  signal.addEventListener('abort', abortPromise, { once:true })
+  const removeAbortListener = () => { signal.removeEventListener('abort', abortPromise) }
+  promise.then(removeAbortListener, removeAbortListener)
+
+  return promise
+}
```

Comment:
We might want to move this to `RateLimitedQueue` itself so it can be used in other plugins as well, not sure what's best.

Reply:
Yeah I have to say it's getting really complex to keep track of all the different upload pieces. We should definitely centralize it. Probably more an issue for #4173 than this PR.